src/connectionController.ts

```diff
@@ -31,6 +24,24 @@ import { StorageVariables } from './storage';
 import type { StatusView } from './views';
 import type TelemetryService from './telemetry/telemetryService';
 import LINKS from './utils/links';
+import type {
+  ConnectionInfo as ConnectionInfoFromLegacyDS,
+  ConnectionOptions as ConnectionOptionsFromLegacyDS,
+} from 'mongodb-data-service-legacy';
+import {
+  getConnectionTitle,
+  extractSecrets,
+  mergeSecrets,
+  convertConnectionModelToInfo,
```

Comment:
We could copy the source code for this one, but then we’d still have a dependency on `mongodb-connection-model` and would need to use the outdated version of that (or, we could, recursively, also add the source code for *that* entire package).